services/blockchain-indexer/shared/indexer/transactionProcessor/token/transferCrossChain.js

```diff
@@ -20,31 +20,36 @@ const {
 const config = require('../../../../config');
 
 const { TRANSACTION_STATUS } = require('../../../constants');
-const { indexAccountByAddress } = require('../../accountIndex');
 
 const logger = Logger();
 
 const MYSQL_ENDPOINT = config.endpoints.mysql;
 
+const accountsTableSchema = require('../../../database/schema/accounts');
 const transactionsTableSchema = require('../../../database/schema/transactions');
 
+const getAccountsTable = () => getTableInstance(accountsTableSchema, MYSQL_ENDPOINT);
 const getTransactionsTable = () => getTableInstance(transactionsTableSchema, MYSQL_ENDPOINT);
 
 // Command specific constants
 const COMMAND_NAME = 'transferCrossChain';
 
 // eslint-disable-next-line no-unused-vars
 const applyTransaction = async (blockHeader, tx, events, dbTrx) => {
-	logger.trace(`Updating index for the account with address ${tx.params.recipientAddress} asynchronously.`);
-	indexAccountByAddress(tx.params.recipientAddress);
-
-	if (tx.executionStatus !== TRANSACTION_STATUS.SUCCESS) return;
-
 	tx = {
 		...tx,
 		...tx.params,
 	};
 
+	const accountsTable = await getAccountsTable();
+	// Update the account balance
+	const account = { address: tx.recipientAddress };
+	logger.trace(`Updating account index for the account with address ${account.address}.`);
+	await accountsTable.upsert(account, dbTrx);
+	logger.debug(`Updated account index for the account with address ${account.address}.`);
+
+	if (tx.executionStatus !== TRANSACTION_STATUS.SUCCESS) return;
+
```

Comment:
Did this changes because `indexAccountByAddress` function is not working. Should we remove it?

Same for other functions.

Reply:
As discussed offline, please update the implementation for `indexAccountByAddress` and others. Please revert the transaction processor changes.